File: app/src/main/java/me/srikavin/quiz/model/AnswerResponse.java
package me.srikavin.quiz.model;

import java.util.List;

/**
 * A Model class that holds the structure of an answer response as returned by a repository
 */
public class AnswerResponse {
    private boolean correct;
    private QuizQuestion question;
    private List<QuizAnswer> correctAnswers;


    public AnswerResponse(boolean correct, QuizQuestion question, List<QuizAnswer> correctAnswers) {
        this.correct = correct;
        this.question = question;
        this.correctAnswers = correctAnswers;
    }

    public boolean isCorrect() {
        return correct;
    }

    public List<QuizAnswer> getCorrectAnswers() {
        return correctAnswers;
    }
}

File: app/src/main/java/me/srikavin/quiz/model/AnswerResponse.kt
package me.srikavin.quiz.model

import me.srikavin.quiz.network.common.model.data.QuizAnswerModel
import me.srikavin.quiz.network.common.model.data.QuizQuestionModel

/**
 * A Model class that holds the structure of an answer response as returned by a repository
 */
class AnswerResponse(
        /**
         * Represents whether the answer to the last question was correct
         */
        val isCorrect: Boolean,
        /**
         * The [QuizQuestionModel] representing the last question
         */
        val question: QuizQuestionModel,
        /**
         * A list of [QuizAnswerModel] containing all of the correct answers
         */
        val correctAnswers: List<QuizAnswerModel>
)
